client/web/web.go

```diff
@@ -433,6 +433,183 @@ func (s *Server) serveLoginAPI(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+type apiHandler[data any] struct {
+	s *Server
+	w http.ResponseWriter
+	r *http.Request
+
+	// peerAllowed must be set through either invoking
+	//
+	// 1. apiHandler.allPeersAllowed to declare that all peers are allowed
+	//    to execute this handler regardless of peer capabilities
+	//
+	// or 2. apiHandler.restrictPeers to provide a function for evaluating
+	//       a peer's access to this handler.
+	//
+	// A request will fail with error if one of these two functions has not
+	// been invoked first.
+	peerAllowed func(data data, peer peerCapabilities) bool
+}
+
+func newHandler[data any](s *Server, w http.ResponseWriter, r *http.Request) *apiHandler[data] {
+	return &apiHandler[data]{s: s, w: w, r: r}
+}
+
+func (a *apiHandler[data]) allPeersAllowed() *apiHandler[data] {
+	a.peerAllowed = func(_ data, _ peerCapabilities) bool { return true }
```

Comment:
Would it make sense to have this set / passed into `newHandler` and have `allPeersAllowed` be a var that can be used for that argument? Mostly wondering since that would force the value to be explicitly set and error on compile time if the arg is missing.

Reply:
Oh yeah, I like that idea - will update

Reply:
Updated!